front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -55,38 +77,24 @@ async function handler(
           },
         });
       }
+      const bodyValidation = PostManagedDataSourceRequestBodySchema.decode(
+        req.body
+      );
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
 
-      // retrieve suffix GET parameter
-      let suffix: string | null = null;
-      if (req.query.suffix && typeof req.query.suffix === "string") {
-        suffix = req.query.suffix;
-      }
-      if (suffix && !/^[a-z0-9\-_]{1,16}$/.test(suffix)) {
         return apiError(req, res, {
           status_code: 400,
           api_error: {
             type: "invalid_request_error",
-            message: "Invalid suffix.",
+            message: `Invalid request body: ${pathError}`,
           },
         });
       }
 
-      if (!req.body || typeof req.body.connectionId !== "string") {
-        return apiError(req, res, {
-          status_code: 400,
-          api_error: {
-            type: "invalid_request_error",
-            message: "The request body is missing.",
-          },
-        });
-      }
+      const { type, connectionId, url, provider } = bodyValidation.right;
 
-      if (
-        !req.body.provider ||
-        !["slack", "notion", "github", "google_drive", "intercom"].includes(
-          req.body.provider
-        )
-      ) {
+      if (!CONNECTOR_PROVIDERS.includes(provider as ConnectorProvider)) {
```

Comment:
Would IMO look cleaner with [this](https://github.com/dust-tt/dust/blob/c66ba3febcef280c2235b3a210a9bbd632c93df8/connectors/src/types/connector.ts#L3)

Could be moved to types